hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/handlers/MessageReadHandler.java

```diff
@@ -210,6 +213,13 @@ private void endWithoutDefaultResponse(HttpServerExchange exchange) {
         @Override
         public boolean handleDefaultResponse(HttpServerExchange exchange) {
             if (exchange.getAttachment(AttachmentContent.KEY).isResponseReady()) {
+                if (exchange.getStatusCode() == 200) {
+                    try {
+                        exchange.setStatusCode(500);
+                    } catch (RuntimeException e) {
+                        logger.warn("Exception has been thrown during an exchange status modification", e);
```

Comment:
As fa as I understand, `setStatusCode` can throw an exception only if response was already sent (IllegalStateException). In such case client would receive 200 response which is not documented and invalid for Hermes. There is nothing we can do about it in this place however I would use `error` log level so maybe someone will notice it in logs.